src/scope/new/base.js

```diff
@@ -0,0 +1,198 @@
+'use strict'
+
+let id = 0
+
+/**
+ * The Datadog Scope Manager. This is used for context propagation.
+ *
+ * @hideconstructor
+ */
+class Scope {
+  /**
+   * Get the current active span or null if there is none.
+   *
+   * @returns {Span} The active span.
+   */
+  active () {
+    return this._active() || null
+  }
+
+  /**
+   * Activate a span in the scope of a function.
+   *
+   * @param {external:"opentracing.Span"} span The span to activate.
+   * @param {Function} [callback] Function that will have the span activated on its scope.
+   * @returns The return value of the callback.
+   */
+  activate (span, callback) {
+    if (typeof callback !== 'function') return callback
```

Comment:
When would this happen? Are there alternatives for `callback` ?

Reply:
`scope.activate(span, 'boom')`

Reply:
But is this something someone would actually do?

Reply:
If they do, I expect the scope manager to now throw.

Reply:
:+1: